eolearn/mask/cloud_mask.py

```diff
@@ -89,81 +87,29 @@ def __init__(
         if dilation_size is not None and dilation_size > 0:
             self.dil_kernel = disk(dilation_size).astype(np.uint8)
 
-        self._classifier: ClassifierType | Booster | None = None
-
-    @property
-    def classifier(self) -> ClassifierType | Booster:
-        """An instance of a custom-provided cloud classifier. Loaded only the first time it is required."""
-        if self._classifier is None:
-            path = os.path.join(self.MODELS_FOLDER, self.CLASSIFIER_NAME)
-            self._classifier = Booster(model_file=path)
-
-        return self._classifier
-
-    @staticmethod
-    def _run_prediction(classifier: ClassifierType | Booster, features: np.ndarray) -> np.ndarray:
-        """Uses classifier object on given data"""
-        is_booster = isinstance(classifier, Booster)
-
-        predict_method = classifier.predict if is_booster else classifier.predict_proba
-        prediction: np.ndarray = execute_with_mp_lock(predict_method, features)
-
-        return prediction if is_booster else prediction[..., 1]
-
-    def _average(self, data: np.ndarray) -> np.ndarray:
-        return cv2.filter2D(data.astype(np.float64), -1, self.avg_kernel, borderType=cv2.BORDER_REFLECT)
-
-    def _dilate(self, data: np.ndarray) -> np.ndarray:
-        return (cv2.dilate(data.astype(np.uint8), self.dil_kernel) > 0).astype(np.uint8)
-
-    def _average_all(self, data: np.ndarray) -> np.ndarray:
-        """Average over each spatial slice of data"""
-        if self.avg_kernel is not None:
-            return _apply_to_spatial_axes(self._average, data, (1, 2))
-
-        return data
-
-    def _dilate_all(self, data: np.ndarray) -> np.ndarray:
-        """Dilate over each spatial slice of data"""
-        if self.dil_kernel is not None:
-            return _apply_to_spatial_axes(self._dilate, data, (1, 2))
-
-        return data
-
-    def _do_single_temporal_cloud_detection(self, bands: np.ndarray) -> np.ndarray:
-        """Performs a cloud detection process on each scene separately"""
-        output_proba = []
-        _, height, width, n_bands = bands.shape
-
-        for img in bands:
-            features = img.reshape(height * width, n_bands)
-            proba = self._run_prediction(self.classifier, features)
-            output_proba.append(proba.reshape(height, width, 1))
-
-        return np.array(output_proba)
+        self.classifier = S2PixelCloudDetector(
```

Comment:
just wondering if this should still be lazy loaded (pickling etc.)

Reply:
I believe it should be handled in `S2PixelCloudDetector`, because in the `init` phase (see [L60](https://github.com/sentinel-hub/sentinel2-cloud-detector/blob/master/s2cloudless/cloud_detector.py#L60)) the classifier is still not loaded, and is only loaded when the `classifier` method is called, which happens during the `execute` phase of the new task.

Reply:
we really are smart sometimes :thinking: